Source/WTF/wtf/cocoa/FileSystemCocoa.mm

```diff
@@ -239,6 +243,20 @@ bool setExcludedFromBackup(const String& path, bool excluded)
     return true;
 }
 
+bool markPurgeable(const String& path)
+{
+    CString fileSystemPath = fileSystemRepresentation(path);
+    if (fileSystemPath.isNull())
+        return false;
+
+#if HAVE(APFS_CACHEDELETE_PURGEABLE)
+    uint64_t flags = APFS_MARK_PURGEABLE | APFS_PURGEABLE_DATA_TYPE | APFS_PURGEABLE_MARK_CHILDREN;
+    return fsctl(fileSystemPath.data(), APFSIOC_MARK_PURGEABLE, &flags, 0) != -1;
+#endif
+
+    return false;
```

Comment:
Would be good for this to be in a `#else` case. Weird to have two return statements otherwise.